dev/com.ibm.ws.microprofile.metrics.cdi.3.0/src/io/astefanutti/metrics/cdi30/MetricResolver.java

```diff
@@ -0,0 +1,481 @@
+    }
+
+    protected <E extends Member & AnnotatedElement, T extends Annotation> Of<T> elementResolverOf(E element, Class<T> metric) {
+        T annotation = element.getAnnotation(metric);
+
+        // See if we have the name cached
+        String name = extension.getMetricNameForMember(element, annotation);
+
+        boolean initialDiscovery = false;
+        if (name == null) {
+            name = metricName(element, metric, metricName(annotation), isMetricAbsolute(annotation));
+            initialDiscovery = true;
+        }
+
+        MetadataBuilder mdb = Metadata.builder().withName(name).withType(this.getType(annotation)).withUnit(this.getUnit(annotation)).withDescription(this.getDescription(annotation)).withDisplayName(this.getDisplayname(annotation));
+
+        String[] tags = this.getTags(annotation);
+
+        Of<T> of = new DoesHaveMetric<>(annotation, name, mdb.build(), initialDiscovery, tags);
+        checkReusable(of);
+        return of;
+    }
+
+    protected <T extends Annotation> boolean hasMetricAnnotationBean(Class<T> metric, Class<?> bean) {
+        if (bean.isAnnotationPresent(metric))
+            return true;
+
+        /*
+         * Go through the annotations available in this class to see if a sterotype is present
+         * that contains the metric annotation we are testing for
+         */
+        Annotation[] beanAnotations = bean.getAnnotations();
+        for (Annotation annotation : beanAnotations) {
+
+            Class<? extends Annotation> annotationType = annotation.annotationType();
+            if (beanManager.isStereotype(annotationType) && annotationType.isAnnotationPresent(metric)) {
+                return true;
+            }
+        }
+
+        return false;
+    }
+
+    protected <T extends Annotation> T getAnnotationBean(Class<T> metric, Class<?> bean) {
+        T annotationObj = bean.getAnnotation(metric);
+        if (annotationObj != null)
+            return annotationObj;
+
+        /*
+         * Go through the annotations available in this class to see if a sterotype is present
+         * that contains the metric annotation we are testing for and retrieves the Annotation
+         */
+
+        Annotation[] beanAnotations = bean.getAnnotations();
+        for (Annotation annotation : beanAnotations) {
+
+            Class<? extends Annotation> annotationType = annotation.annotationType();
+            if (beanManager.isStereotype(annotationType) && annotationType.isAnnotationPresent(metric)) {
+                annotationObj = annotationType.getAnnotation(metric);
+                if (annotationObj != null) {
+                    return annotationObj;
+                }
+            }
+        }
+
+        return null;
+    }
+
+    protected <E extends Member & AnnotatedElement, T extends Annotation> Of<T> beanResolverOf(E element, Class<T> metric, Class<?> bean) {
+
+        if (hasMetricAnnotationBean(metric, bean)) {
```

Comment:
166-167 using the above two new methods.
It use to directly check the bean if the annotation was present and retrieve it. That of course did not support stereotypes.